pkg/antctl/raw/multicluster/get/member_token.go

```diff
@@ -107,41 +103,41 @@ func runEToken(cmd *cobra.Command, args []string) error {
 			return err
 		}
 
-		memberTokens = append(memberTokens, memberToken)
+		outToken := &memberToken
+		if optionsToken.outputFormat != "" {
+			outToken = common.ConvertMemberTokenSecret(outToken)
+		}
+		return output(*outToken, true, optionsToken.outputFormat, cmd.OutOrStdout(), membertoken.Transform)
 	} else {
 		memberTokenList := &corev1.SecretList{}
 		err = optionsToken.k8sClient.List(context.TODO(), memberTokenList, &client.ListOptions{Namespace: optionsToken.namespace})
 		if err != nil {
 			return err
 		}
-		memberTokens = memberTokenList.Items
-	}
-
-	opaqueMemberTokens := []corev1.Secret{}
-	for _, memberToken := range memberTokens {
-		if memberToken.Annotations[common.CreateByAntctlAnnotation] == "true" {
-			opaqueToken := common.ConvertMemberTokenSecret(memberToken)
-			opaqueMemberTokens = append(opaqueMemberTokens, opaqueToken)
+		opaqueMemberTokens := []corev1.Secret{}
+		for _, memberToken := range memberTokenList.Items {
+			// Ignore tokens not created by antctl mc command.
+			if memberToken.Annotations[common.CreateByAntctlAnnotation] == "true" {
+				outToken := &memberToken
+				if optionsToken.outputFormat != "" {
+					outToken = common.ConvertMemberTokenSecret(outToken)
+					if optionsToken.namespace == "" {
+						// ConvertMemberTokenSecret() does not set Namespace of the Secret.
+						outToken.Namespace = memberToken.Namespace
+					}
+				}
+				opaqueMemberTokens = append(opaqueMemberTokens, *outToken)
+			}
 		}
-	}
 
-	if len(opaqueMemberTokens) == 0 {
-		if singleToken {
-			return fmt.Errorf("Member token %s created by antctl is not found", args[0])
+		if len(opaqueMemberTokens) == 0 {
+			if optionsToken.namespace != "" {
+				fmt.Fprintf(cmd.ErrOrStderr(), "No token found in Namespace %s\n", optionsToken.namespace)
+			} else {
+				fmt.Fprintln(cmd.ErrOrStderr(), "No token found")
+			}
```

Comment:
namespace check can be removed since you already handled single token case before.
```suggestion
		fmt.Fprintln(cmd.ErrOrStderr(), "No token found")
```

Reply:
I feel no issue to print the Namespace if Namespace is provided. But sure I can remove it if you prefer.

Reply:
Just checked other commands and found they all print Namespace when Namespace is provided. I would keep this command the same to avoid more changes.

Reply:
You are right, I misread the codes.